operator/pkg/securedcluster/scanner/auto_sense.go

```diff
@@ -21,14 +30,39 @@ func AutoSenseLocalScannerSupport(ctx context.Context, client ctrlClient.Client,
 		if err != nil {
 			return false, errors.Wrap(err, "detecting presence of a Central CR in the same namespace")
 		}
-		return !siblingCentralPresent, nil
+		isOpenShift, err := isRunningOnOpenShift(ctx, client, s.GetNamespace())
+		if err != nil {
+			return false, errors.Wrap(err, "cannot fetch OpenShift ClusterVersion resource")
+		}
+		enableScanner := isOpenShift && !siblingCentralPresent
+		return enableScanner, nil
 	case platform.LocalScannerComponentDisabled:
 		return false, nil
 	}
 
 	return false, errors.Errorf("invalid spec.scanner.scannerComponent %q", scannerComponent)
 }
 
+func isRunningOnOpenShift(ctx context.Context, client ctrlClient.Client, namespace string) (bool, error) {
+	log := ctrlLog.FromContext(ctx)
+
+	clusterVersion := &osconfigv1.ClusterVersion{}
+	key := ctrlClient.ObjectKey{Namespace: namespace, Name: ClusterVersionDefaultName}
+	err := client.Get(ctx, key, clusterVersion)
+	if err != nil && errorsK8s.IsNotFound(err) {
+		log.Error(err, "OpenShift ClusterVersion not found")
```

Comment:
Imho this is not an error because we expect it on Kubernetes.

```suggestion
		log.Info(err, "Running on Kubernetes, OpenShift ClusterVersion was not found")
```